### PR TITLE
SCP-2863: refactoring set_range to account for missing/na values

### PR DESCRIPTION
**Pull request by bistline**
As discovered in many [bug](https://sentry.io/organizations/broad-institute/issues/1997315121/?project=1424198&query=is%3Aunresolved) reports, issues in parsing or user error in generating data can lead to errors when attempting to compute ranges for scatter plots to enforce square/cube aspect ratios.  This fix now accounts for `NaN` or `nil` values better and rejects them accordingly.  The `SiteController#set_range` method has also been refactored for better readability.

This PR satisfies SCP-2863

### CODE REVIEW

File: app/controllers/site_controller.rb [hunk 1]
```diff
@@ -2024,13 +2024,20 @@ def set_range(inputs)
...
+      end
+      aggregate_range = raw_values.flatten.minmax
       # add 2% padding to range
-      scope = (@vals.first - @vals.last) * 0.02
-      raw_range = [@vals.first + scope, @vals.last - scope]
-      range[:x] = raw_range
-      range[:y] = raw_range
-      range[:z] = raw_range
+      scope = (aggregate_range.first - aggregate_range.last) * 0.02
+      absolute_range = [aggregate_range.first + scope, aggregate_range.last - scope]
```
Comment (eweitz) [inline app/controllers/site_controller.rb:2037]: Clearer naming:

```suggestion
      padding = (aggregate_range.first - aggregate_range.last) * 0.02
      absolute_range = [aggregate_range.first + padding, aggregate_range.last - padding]
```

File: app/controllers/site_controller.rb [hunk 2]
```diff
@@ -2024,13 +2024,20 @@ def set_range(inputs)
       range = @cluster.domain_ranges
     else
       # take the minmax of each domain across all groups, then the global minmax
-      @vals = inputs.map {|v| domain_keys.map {|k| RequestUtils.get_minmax(v[k])}}.flatten.minmax
+      raw_values = []
+      inputs.each do |input|
+        domain_keys.each do |domain|
+          domain_range = RequestUtils.get_minmax(input[domain])
+          raw_values << domain_range if domain_range.any?
```
Comment (eweitz) [inline app/controllers/site_controller.rb:2031]: Please add a brief code comment to note this accounts for missing/na values.

**Review by eweitz — APPROVED** [review]
Looks good!  I noted two nits.

File: bin/load_env_secrets.sh [hunk 3]
```diff
@@ -81,7 +81,13 @@ if [[ -n $VAULT_SECRET_PATH ]] ; then
   do
     echo "setting value for: $key"
     curr_val=$(echo $VALS | jq .data | jq --raw-output .$key)
-    export $key=$curr_val
+    # honor PORTAL_NAMESPACE from the environment, if present
```
Comment (devonbush) [inline bin/load_env_secrets.sh:84]: are we quite sure there aren't any lingering PORTAL_NAMESPACE variables in the prod environment?  I assume you've thought of this, but wanted to double-check :)

**Review by devonbush — APPROVED** [review]

File: bin/load_env_secrets.sh (hunk 3 above)
Comment (bistline) [reply bin/load_env_secrets.sh:84]: `$PORTAL_NAMESPACE` is the only variable that is defined both in `vault` and can be passed into `load_env_secrets.sh`.  Everything else is sourced from `vault`, or passed only as an argument (like `$PASSENGER_APP_ENV`).

**Pull request merged**